### PR TITLE
Josh Salyer

### PR DESCRIPTION
**Pull request by jsalyer23**

### CODE REVIEW

File: lib/all_unique.js [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
\ No newline at end of file
+}
+
+/* Algorithmic Thought Process:
+  First check if there is only one character in the string. If there's only one
+  character in the string then it is not repeated so return the string. If there
+  is more than one character in the string, go through each character and compare
+  the current character with each other character in the string. If the two characters
+  that are being compared are the same, return false. If there isn't any matching
+  characters in the string, return true.
```
Comment (halfghaninne) [inline lib/all_unique.js:23]: This thought process seems sound! Let me know if you have specific questions.

**Comment by jsalyer23** [comment]
I don't understand why I'm getting this error because I don't see anywhere }); even shows up in my code. This is the error I'm getting back:
/home/ubuntu/workspace/lib/all_unique.js:25
});
 ^

SyntaxError: Unexpected token )
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/ubuntu/workspace/test/all_unique_test.js:3:18)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at /home/ubuntu/workspace/node_modules/mocha/lib/mocha.js:220:27
    at Array.forEach (native)
    at Mocha.loadFiles (/home/ubuntu/workspace/node_modules/mocha/lib/mocha.js:217:14)
    at Mocha.run (/home/ubuntu/workspace/node_modules/mocha/lib/mocha.js:469:10)
    at Object.<anonymous> (/home/ubuntu/workspace/node_modules/mocha/bin/_mocha:404:18)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at startup (node.js:139:18)
    at node.js:968:3

File: lib/all_unique.js [hunk 2]
```diff
@@ -1,8 +1,18 @@
 module.exports = {
   all_unique: function(str){
-  
+  var separatedString = str.slice(0);
```
Comment (halfghaninne) [inline lib/all_unique.js]: It's very hard to look for where that error might be coming from without proper indentation. Please indent and if you are still having issues, commit and push again.

File: lib/all_unique.js [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
+  }
+  if (equalCharCount === 0)
+    return true;
+  else
+    return false;
+    }
   }
-}
 
 /* Algorithmic Thought Process:
```
Comment (halfghaninne) [inline lib/all_unique.js:17]: Code comments in JS begin with `//`

Comment (jsalyer23) [inline lib/all_unique.js:17]: [Multi line comments](http://www.w3schools.com/js/js_comments.asp) are written using /\* and */. At least that's what I've read. I'll try improving indentation and commit again.

Comment (halfghaninne) [inline lib/all_unique.js:17]: Eep. Good call.

**Comment by halfghaninne** [comment]
👍

**Pull request closed without merging**